src/node_file.cc

```diff
@@ -1193,63 +1152,34 @@ static void Stat(const FunctionCallbackInfo<Value>& args) {
       env, permission::PermissionScope::kFileSystemRead, path.ToStringView());
 
   bool use_bigint = args[1]->IsTrue();
-  FSReqBase* req_wrap_async = GetReqWrap(args, 2, use_bigint);
-  if (req_wrap_async != nullptr) {  // stat(path, use_bigint, req)
+  if (argc == 3) {  // stat(path, use_bigint, req)
+    FSReqBase* req_wrap_async = GetReqWrap(args, 2, use_bigint);
+    CHECK_NOT_NULL(req_wrap_async);
     FS_ASYNC_TRACE_BEGIN1(
         UV_FS_STAT, req_wrap_async, "path", TRACE_STR_COPY(*path))
     AsyncCall(env, req_wrap_async, args, "stat", UTF8, AfterStat,
               uv_fs_stat, *path);
-  } else {  // stat(path, use_bigint, undefined, ctx)
-    CHECK_EQ(argc, 4);
-    FSReqWrapSync req_wrap_sync;
+  } else {  // stat(path, use_bigint, undefined, do_not_throw_if_no_entry)
+    bool do_not_throw_if_no_entry = args[3]->IsFalse();
+    FSReqWrapSync req_wrap_sync("stat", *path);
     FS_SYNC_TRACE_BEGIN(stat);
-    int err = SyncCall(env, args[3], &req_wrap_sync, "stat", uv_fs_stat, *path);
+    int result;
+    if (do_not_throw_if_no_entry) {
+      result = SyncCallAndThrowIf(
+          is_uv_error_except_no_entry, env, &req_wrap_sync, uv_fs_stat, *path);
+    } else {
+      result = SyncCallAndThrowOnError(env, &req_wrap_sync, uv_fs_stat, *path);
+    }
     FS_SYNC_TRACE_END(stat);
-    if (err != 0) {
-      return;  // error info is in ctx
+    if (result < 0) {
```

Comment:
You added `is_uv_error` function, but didn't use it. Maybe we should remove that function anywhere except the SyncCall

Reply:
is_uv_error is used in SyncCallAndThrowOnErro as a parameter.

Reply:
since it's only used in that function, maybe we can remove that function, or use it in all functions in `node_file`?

Reply:
I reused that for the error checking branches in node_file.cc.